### PR TITLE
[FIX] runbot_travis2docker: Avoid request if the token and url is equal

### PR DESCRIPTION
**Pull request by JesusZapata**
Fix https://github.com/Vauxoo/runbot-addons/issues/117

When the url of weblate and the token is equal are stored the projects for used on another branch
This avoid many request to the same weblate API

### CODE REVIEW

**Comment by JesusZapata** [comment]
@moylop260 
What do you think about this change?

**Comment by JesusZapata** [comment]
@moylop260 
I made one function `read_group` what get the projects and the components for each project.
I made a better use of the cache

File: runbot_travis2docker/models/runbot_branch.py [hunk 1]
```diff
@@ -5,12 +5,47 @@
...
 import re
 
-import requests
 import subprocess
+import requests
 
 from openerp import fields, models, api
 
 
+def read_group(branch):
```
Comment (moylop260) [inline runbot_travis2docker/models/runbot_branch.py]: Better use `ormcache`
Use the following PR to get a example of the `How to`:
https://github.com/Vauxoo/enterprise/pull/35

Comment (JesusZapata) [reply runbot_travis2docker/models/runbot_branch.py]: @moylop260 
Thanks for teach me the annotation `@tools.ormcache` it is very powerfull

**Comment by JesusZapata** [comment]
@moylop260 
This PR reduce the API request if the `weblate_url` and `weblate_token` is equal

**For example:**
If you have 70 branches with `uses_weblate` in True.
And all `repo_id` values `weblate_url` and `weblate_token` are the same values

| Total request now | Total request in this PR |
| --------- | ---------- |
| 280 req | 1 req |

In this PR only once is made the request the next time the annotation ormcache returned the stored value

This optimization is extremely useful

File: runbot_travis2docker/models/runbot_branch.py [hunk 2]
```diff
@@ -33,43 +33,52 @@ def _compute_name_weblate(self):
                         dict(match.groupdict(), branch=branch['branch_name']))
             branch.name_weblate = name
 
+    @tools.ormcache('url', 'token')
+    def get_weblate_projects(self, url, token):
+        """Search all weblate API, project and component information.
+        Store into the local variable 'read_group.projects' to use as cache"""
```
Comment (moylop260) [inline runbot_travis2docker/models/runbot_branch.py]: Update docstring

File: runbot_travis2docker/models/runbot_branch.py [hunk 3]
```diff
@@ -97,6 +106,7 @@ def cron_weblate(self):
                         continue
                     branch.force_weblate()
                     updated_branch = component['branch']
+        self.clear_caches()
```
Comment (moylop260) [inline runbot_travis2docker/models/runbot_branch.py]: Add a comment of why is required this clear cache at end method

File: runbot_travis2docker/models/runbot_branch.py (hunk 2 above)
Comment (JesusZapata) [reply runbot_travis2docker/models/runbot_branch.py]: @moylop260 
Done

File: runbot_travis2docker/models/runbot_branch.py (hunk 3 above)
Comment (JesusZapata) [reply runbot_travis2docker/models/runbot_branch.py]: @moylop260 
Done

**Pull request merged**